src/test/java/com/google/api/generator/engine/writer/JavaWriterVisitorTest.java

```diff
@@ -98,6 +100,33 @@ public void writeAnnotation_withDescription() {
     assertEquals(writerVisitor.write(), "@SuppressWarnings(\"all\")\n");
   }
 
+  @Test
+  public void writeNewObjectValue_withArgs() {
+    ConcreteReference reference = ConcreteReference.withClazz(Integer.class);
+    TypeNode type = TypeNode.withReference(reference);
+    ValueExpr valueExpr =
+        ValueExpr.builder()
+            .setValue(PrimitiveValue.builder().setType(TypeNode.INT).setValue("123").build())
+            .build();
+    NewObjectValue newObjectValue =
+        NewObjectValue.builder().setType(type).setArguments(Arrays.asList(valueExpr)).build();
+    newObjectValue.accept(writerVisitor);
+    assertEquals(writerVisitor.write(), "new Integer(123)");
+  }
+
+  @Test
+  public void writeNewObjectValue_withGenerics() {
+    ConcreteReference reference =
+        ConcreteReference.builder()
+            .setClazz(LinkedList.class)
+            .setGenerics(Arrays.asList(ConcreteReference.withClazz(Object.class)))
+            .build();
+    TypeNode type = TypeNode.withReference(reference);
+    NewObjectValue newObjectValue = NewObjectValue.genericBuilder().setType(type).build();
+    newObjectValue.accept(writerVisitor);
+    assertEquals(writerVisitor.write(), "new LinkedList<Object>()");
+  }
```

Comment:
Could we please add a test case that uses both generics (more than one generic) and arguments (more than one, and not just ValueExpr, e.g. a method invocation expr or variable)?